lib/ansible/plugins/action/__init__.py

```diff
@@ -311,18 +311,23 @@ def _fixup_perms(self, remote_path, remote_user, execute=False, recursive=True):
         if self._play_context.become and self._play_context.become_user not in ('root', remote_user):
             # Unprivileged user that's different than the ssh user.  Let's get
             # to work!
-            if remote_user == 'root':
-                # SSh'ing as root, therefore we can chown
-                res = self._remote_chown(remote_path, self._play_context.become_user, recursive=recursive)
-                if res['rc'] != 0:
-                    raise AnsibleError('Failed to set owner on remote files (rc: {0}, err: {1})'.format(res['rc'], res['stderr']))
+
+            # Try chown'ing the file. This will only work if our SSH user has
+            # root privileges, but since we can't reliably determine that from
+            # the username (think "toor" on FreeBSD), let's just try first and
+            # apologize later:
+            res = self._remote_chown(remote_path, self._play_context.become_user, recursive=recursive)
+            if res['rc'] == 0:
+                # Only continue with chmod if chown worked
```

Comment:
@abadger So you'd prefer something like
```
-                # Only continue with chmod if chown worked
-                if execute:
-                     # root can read things that don't have read bit but can't
-                     # execute them.
+                # root can read things that don't have read bit but can't
+                # execute them, so we might need to set the execute bit if
+                # we're root. We just ran chown successfully, so apparently
+                # we are root.
+                if execute:
```
?
That does sound more useful than my original version.